adapters/adview/adview.go

```diff
@@ -0,0 +1,144 @@
+package adview
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+// Builder builds a new instance of the adview adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	endpointTemplate, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: endpointTemplate,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var bidderExt adapters.ExtImpBidder
+	imp := &request.Imp[0]
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		errors = append(errors, &errortypes.BadInput{
+			Message: fmt.Sprintf("invalid imp.ext, %s", err),
+		})
+		return nil, append(errors, err)
+	}
+	//use adview
+	var advImpExt openrtb_ext.ExtImpAdView
+	if err := json.Unmarshal(bidderExt.Bidder, &advImpExt); err != nil {
+		errors = append(errors, &errortypes.BadInput{
+			Message: fmt.Sprintf("invalid bidderExt.Bidder, %s", err),
+		})
+		return nil, append(errors, err)
```

Comment:
Same comment as above. It looks like we are adding the error to `errors` twice here.